legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java

```diff
@@ -0,0 +1,370 @@
+import org.jboss.as.controller.registry.OperationEntry;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.messaging.logging.MessagingLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.dmr.Property;
+
+/**
+ * Operation to migrate from the legacy messaging subsystem to the new messaging-activemq subsystem.
+ *
+ * This operation must be performed when the server is in admin-only mode.
+ * Internally, the operation:
+ *
+ * <ul>
+ *     <li>query the description of all the messaging subsystem by invoking the :describe operation.
+ *     This returns a list of :add operations for each messaging resources.</li>
+ *     <li>:add the new org.widlfy.extension.messaging-activemq extension</li>
+ *     <li>for each messaging resources, transform the :add operations to add the
+ *     corresponding resource to the new messaging-activemq subsystem.
+ *     In this step, changes to the resources model are taken into account
+ *     (e.g. cluster-connection's connector-ref is named connector-name in the new messaging-activemq subsystem.)</li>
+ *     <li>:remove the messaging subsystem</li>
+ * </ul>
+ *
+ * The companion <code>:describe-migration</code> operation will return a list of all the actual operations that would be
+ * performed during the invocation of the <code>:migrate</code> operation.
+ *
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2015 Red Hat inc.
+ */
+
+public class MigrateOperation implements OperationStepHandler {
+
+    private static final String MESSAGING_ACTIVEMQ_EXTENSION = "org.wildfly.extension.messaging-activemq";
+    private static final String MESSAGING_ACTIVEMQ_MODULE = "org.wildfly.extension.messaging-activemq";
+
+    private static final String NEW_ENTRY_SUFFIX = "-new";
+
+    private static final OperationStepHandler DESCRIBE_MIGRATION_INSTANCE = new MigrateOperation(true);
+    private static final OperationStepHandler MIGRATE_INSTANCE = new MigrateOperation(false);
+
+    private static final AttributeDefinition ADD_LEGACY_ENTRIES = SimpleAttributeDefinitionBuilder.create("add-legacy-entries", BOOLEAN)
+            .setDefaultValue(new ModelNode(false))
+            .build();
+    private static final AttributeDefinition DRY_RUN = SimpleAttributeDefinitionBuilder.create("dry-run", BOOLEAN)
+            .setDefaultValue(new ModelNode(true))
+            .build();
+
+    private final boolean describe;
+
+    private MigrateOperation(boolean describe) {
+
+        this.describe = describe;
+    }
+
+    static void registerOperations(ManagementResourceRegistration registry, ResourceDescriptionResolver resourceDescriptionResolver) {
+        registry.registerOperationHandler(new SimpleOperationDefinitionBuilder("migrate", resourceDescriptionResolver)
+                        .setParameters(ADD_LEGACY_ENTRIES, DRY_RUN)
+                        .setRuntimeOnly()
+                        .setReplyValueType(ModelType.BOOLEAN)
```

Comment:
This should be setReplyType().

But I don't see why this one needs to provide any result at all. The "outcome" = "success" or "failed" part of the response indicates what happened; the "result" looks to just be the "dryRun" param which the caller already knows.